server/service.go

```diff
@@ -243,44 +250,75 @@ func (m *BoostService) handleStatus(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+type (
+	payloadByValidator map[relay.ValidatorPublicKey]types.SignedValidatorRegistration
+	payloadsByRelay    map[relay.Entry][]types.SignedValidatorRegistration
+)
+
+func (v payloadsByRelay) add(relays relay.List, payload types.SignedValidatorRegistration) {
+	for _, r := range relays {
+		v[r] = append(v[r], payload)
+	}
+}
+
 // handleRegisterValidator - returns 200 if at least one relay returns 200, else 502
 func (m *BoostService) handleRegisterValidator(w http.ResponseWriter, req *http.Request) {
 	log := m.log.WithField("method", "registerValidator")
-	log.Debug("registerValidator")
 
-	payload := []types.SignedValidatorRegistration{}
+	var payload []types.SignedValidatorRegistration
 	if err := DecodeJSON(req.Body, &payload); err != nil {
 		m.respondError(w, http.StatusBadRequest, err.Error())
 		return
 	}
 
 	ua := UserAgent(req.Header.Get("User-Agent"))
-	log = log.WithFields(logrus.Fields{
+	log.WithFields(logrus.Fields{
 		"numRegistrations": len(payload),
 		"ua":               ua,
-	})
+	}).Info("registering validators")
 
-	relayRespCh := make(chan error, len(m.relays))
+	validatorPayloads := make(payloadByValidator, len(payload))
```

Comment:
no need to preallocate map size. could lead to weird results if for some reason a CL client includes multiple registrations for a single validator

Reply:
Fixed